### PR TITLE
Feat: Cart management and Wishlist

### PR DESCRIPTION
**Pull request by shuntzhere**

### CODE REVIEW

File: src/components/CartCard/CartCard.jsx [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+import AddIcon from "@mui/icons-material/Add";
+import RemoveIcon from "@mui/icons-material/Remove";
+import { useCart } from "../../context/CartContext";
+import { useWishlist } from "../../context/WishlistContext";
+
+export const CartCard = ({ _id, img, name, price, description, quantity }) => {
+  const [count, setCount] = useState(quantity);
+  const { cartDispatch } = useCart();
+  const { wishlistDispatch } = useWishlist();
+  // const finalPrice =
```
Comment (Sameony) [inline src/components/CartCard/CartCard.jsx:12]: do remove comments before merging

File: src/components/Category/Category.jsx [hunk 2]
```diff
@@ -1,15 +1,16 @@
...
 
-export const Category = ({ title, color }) => {
+export const Category = ({ title, color, href }) => {
   return (
     <div
       className="category__name endangered"
       style={{ backgroundColor: color }}
     >
-      <a href="#" className="category__name--title">
+      <Link to={`/products?category=${href}`} className="category__name--title">
```
Comment (Sameony) [inline src/components/Category/Category.jsx:11]: i liked how you displayed selected category from featured section

**Review by Sameony — APPROVED** [review]
Nice PR, i learnt how to make work with less contexts, will try to implement that

File: src/components/CartCard/CartCard.jsx [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+  useEffect(() => {
+    count < 1 || isNaN(count) ? setCount(1) : setCount(count);
+    cartDispatch({
+      type: "CART_QUANTITY",
+      payload: { _id: _id, quantity: count },
+    });
+  }, [count]);
+
+  return (
+    <div className="card horizontal-card flex" key={_id}>
```
Comment (kjais1720) [inline src/components/CartCard/CartCard.jsx:23]: the `key` property is not needed here, it is used where the `map` is being called

File: src/components/CartCard/CartCard.jsx [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+      <div className="card-content">
+        <h3 className="card-header">{name}</h3>
+        <p className="card-text">{description}</p>
+        <div className="card-info flex flex-items-center justify-around">
+          <p className="change__cart">{quantity}</p>
+          <h4 className="rating">{price * quantity}</h4>
+        </div>
+        <div className="cart__buttons">
+          <AddIcon
+            style={{ cursor: "pointer" }}
```
Comment (kjais1720) [inline src/components/CartCard/CartCard.jsx:34]: try to avoid inline styles

File: src/components/CartDetail/CartDetail.jsx [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+      </div>
+      <div className="divider" />
+      <div className="flex justify-between">
+        <p className="cart__details--text">Remaining Credit</p>
+        <p className="cart__details--text">{totalCredit - bill}</p>
+      </div>
+      <button
+        className="btn primary-btn"
+        type="button"
+        style={{ backgroundColor: "green", cursor: "pointer" }}
```
Comment (kjais1720) [inline src/components/CartDetail/CartDetail.jsx:35]: here as well

**Review by kjais1720 — APPROVED** [review]
Looks good overall, just a couple of suggestions made, you can refactor them if you have enough time

**Pull request merged**